analyzer/codechecker_analyzer/analyzers/clangsa/ctu_manager.py

```diff
@@ -95,8 +97,39 @@ def merge_ctu_func_maps(ctu_dir, ctu_func_map_file, ctu_temp_fnmap_folder):
             shutil.rmtree(fnmap_dir, ignore_errors=True)
 
 
+def generate_invocation_list(triple_arch, action, source, config, env):
+    """ Generates the invocation for the source file of the current
+    compilation command. Used during on-demand analysis. The invocation list
+    is a mapping from absolute paths of the source files to the parts of the
+    compilation commands that are used to generate from each one of them. """
+
+    triple_arch_dir = Path(config.ctu_dir, triple_arch)
+    # Ensure, that architecture directory exists.
+    triple_arch_dir.mkdir(parents=True, exist_ok=True)
+
+    invocation_list = triple_arch_dir / 'invocation-list.yml'
+
+    source_path = Path(source).resolve()
+
+    cmd = ctu_triple_arch.get_compile_command(action, config, source)
+    # __clang__analyzer__ macro needs to be set in the imported TUs too.
```

Comment:
This is a good observation, I am not sure if we set this in the vanilla CTU mode at all.

Reply:
We do set it with ast-dump-based mode, but we do not if we do it manually (ctu without CodeChecker). Frankly, I did not even know this was a requirement... :astonished: